compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/ClosureDesugar.java

```diff
@@ -540,6 +899,20 @@ public void visit(BLangTypeDefinition typeDef) {
 
     @Override
     public void visit(BLangClassDefinition classDefinition) {
+        if (!classDefinition.isObjectContructorDecl) {
+            result = classDefinition;
+            return;
+        }
+
+        OCEDynamicEnvironmentData oceData = classDefinition.oceEnvData;
+        BVarSymbol mapSymbol = oceData.mapBlockMapSymbol;
+        if (mapSymbol == null) {
+            // This cannot happen.
+            result = classDefinition;
+            return;
+        }
+
+        desugar.visit(classDefinition);
```

Comment:
Can you check here again? we are directly calling visit method. At that moment, env can be something different that is set in the Desugar class. We should call `desugar.rewrite(Node, Env)`